controller/engine_controller.go

```diff
@@ -822,15 +834,18 @@ func (m *EngineMonitor) refresh(engine *longhorn.Engine) error {
 
 	if !isOldVersion {
 		// Cannot continue to start restoration if expansion is not complete
-		if engine.Spec.VolumeSize != engine.Status.CurrentSize {
-			if !engine.Status.IsExpanding {
+		if engine.Spec.VolumeSize > engine.Status.CurrentSize {
```

Comment:
Since we've already dealt with `>`, we should deal with `<` case as well, even we just report an unexpected error.